packages/pharos/src/components/toast/pharos-toaster.ts

```diff
@@ -52,45 +103,72 @@ export class PharosToaster extends PharosElement {
     super.disconnectedCallback && super.disconnectedCallback();
   }
 
-  private _getToastID(id: string | null) {
+  private _getToastID(id: ToastID | null | undefined) {
     return id || `toast_${uuidv4()}`;
   }
 
   private async _openToast(event: Event): Promise<void> {
-    const toastTag = this.localName.split('pharos-toaster')[0] + 'pharos-toast';
-    const toast = document.createElement(toastTag) as PharosToast;
-    const { content, status, id, indefinite } = (<CustomEvent>event).detail;
-
-    toast.innerHTML = content;
-    toast.status = status || DEFAULT_STATUS;
-    toast.id = this._getToastID(id);
-    toast.indefinite = indefinite || DEFAULT_INDEFINITE;
-    this.insertBefore(toast, this.childNodes[0] || null);
+    const { content, status, id, indefinite } = <ToastCreateDetail>(<CustomEvent>event).detail;
+    const toastId = this._getToastID(id);
+
+    this._toasts = [
+      {
+        content,
+        status: status || DEFAULT_STATUS,
+        id: toastId,
+        indefinite: indefinite || DEFAULT_INDEFINITE,
+      },
+      ...this._toasts,
+    ];
+
     await this.updateComplete;
-    toast.focus();
+    (this.renderRoot.querySelector(`#${toastId}`) as HTMLElement)?.focus();
   }
 
   private _updateToast(event: CustomEvent): void {
-    const { content, status, id } = (<CustomEvent>event).detail;
-    const toast = document.getElementById(this._getToastID(id));
-    if (toast) {
-      toast.innerHTML = content;
-      toast.status = status;
-    }
+    const { content, status, id } = <ToastUpdateDetail>(<CustomEvent>event).detail;
+
+    this._toasts = this._toasts.map((toast: ToastDetail): ToastDetail => {
+      if (toast.id === id) {
+        return {
+          ...toast,
+          content: content || toast.content,
+          status: status || toast.status,
+        };
+      } else {
+        return toast;
+      }
+    });
   }
 
   private _closeToast(event: CustomEvent): void {
-    const { id } = (<CustomEvent>event).detail || {};
-    const toast = document.getElementById(this._getToastID(id));
-    if (toast) {
-      this.removeChild(toast);
-    }
+    const { id } = <ToastCloseDetail>(<CustomEvent>event).detail || {};
+    this._toasts = this._toasts.filter((toast) => toast.id !== id);
+  }
+
+  private _renderToast(toast: ToastDetail): TemplateResult {
+    // This is used to properly render any Pharos components present in the supplied content.
+    // unsafeHTML will render content in general, but does not appear to render components.
+    const toastContentElement = document.createElement('div');
+    toastContentElement.innerHTML = toast.content;
+
+    return html`<pharos-toast
+      id="${toast.id}"
+      status="${toast.status}"
+      ?indefinite="${toast.indefinite}"
+    >
+      ${toastContentElement}
+    </pharos-toast>`;
```

Comment:
@Niznikr curious if you might understand more about why this is and if I've missed some trick here—the following did not work, failing to render e.g. a `pharos-link` present in `toast.content`:

```js
    return html`<pharos-toast
      id="${toast.id}"
      status="${toast.status}"
      ?indefinite="${toast.indefinite}"
    >
      ${unsafeHTML(toast.content)}
    </pharos-toast>`;
```

Reply:
Not certain tbh. Believe I ran into this the first time around. The [Lit Discord](https://discord.com/invite/buildWithLit) might be helpful.

Reply:
Interestingly this doesn't happen in [the Lit playground](https://lit.dev/playground/?#gist=46422f0c6361d69614f57d0e66cef633).

Reply:
is this because pharos-link isn't being registered anywhere in the story?

if I add `PharosLink` as a "local" definition in the toast. it does render correctly
```
export class PharosToaster extends ScopedRegistryMixin(PharosElement) {
  static elementDefinitions = {
    'pharos-toast': PharosToast,
    'pharos-link': PharosLink,
  };
```

** just did this to test if it would work, not suggesting this

Reply:
> is this because pharos-link isn't being registered anywhere in the story?

The components are all registered and available to all stories by nature of [this module](https://github.com/ithaka/pharos/blob/develop/packages/pharos/src/test/initComponents.ts) and [its import](https://github.com/ithaka/pharos/blob/develop/.storybook/main/preview.js#L9).

> if I add PharosLink as a "local" definition in the PharosToaster. it does render correctly

To be clear, it already does render correctly currently. The question is whether there's a way to avoid the DOM manipulation here while maintaining the functionality.

The crux being that the supplied content comes from the consumer, and the toast _cannot predict_ what that content will be, and therefore does not want to be in the business of e.g. registering all possible components that could be in that content.

Reply:
in my testing i added this
```
export class PharosToaster extends ScopedRegistryMixin(PharosElement) {
  static elementDefinitions = {
    'pharos-toast': PharosToast,
    'pharos-link': PharosLink,
  };
```

as well as changed this back to 
```
return html`<pharos-toast
      id="${toast.id}"
      status="${toast.status}"
      ?indefinite="${toast.indefinite}"
    >
      ${unsafeHTML(toast.content)}
    </pharos-toast>`;
```
```

Reply:
I have been assuming this is due to the supplied content being put into the `Toast`'s light DOM, but now that you prompt me to think more about it the `Toast` is inside the `Toaster`'s shadow DOM so I am not sure exactly what I should be expecting. The scope seems like it would indeed be the `Toaster`'s registry.

Reply:
(Updated in the [Discord discussion](https://discord.com/channels/1012791295170859069/1065652155257737286) as well) @chrisjbrown and I tracked this down to the scoped registry behavior, which we now understand does not have inheritance—you either get the global registry without isolation, or you get your total isolation with only explicit components allowed. The DOM manipulation style is what allows us to straddle the two.